### PR TITLE
Activity 302 improve settings

### PR DESCRIPTION
**Pull request by nickvergessen**
Required for https://github.com/owncloud/activity/pull/303

### CODE REVIEW

File: lib/public/activity/iextension.php [hunk 1]
```diff
@@ -50,7 +52,11 @@
 	 * If no additional types are to be added false is to be returned
 	 *
 	 * @param string $languageCode
-	 * @return array|false
+	 * @return array|false Array "stringID of the type" => "translated string description for the setting"
```
Comment (nickvergessen) [inline lib/public/activity/iextension.php]: @MorrisJobke how do we document changes? `@changed 8.x.0` ?

Comment (MorrisJobke) [inline lib/public/activity/iextension.php]: I added it after the since tag and wrote "return type changed in 8.x.y"

**Comment by nickvergessen** [comment]
@jancborchardt would be nice if you can test this together with https://github.com/owncloud/activity/pull/303 from activity.
The email checkbox is then disabled for favorites:
personal_-_owncloud_-_2015-07-03_09 25 49


Others, maybe @MorrisJobke

File: lib/public/activity/iextension.php [hunk 2]
```diff
@@ -50,8 +52,13 @@
...
+	 * 				or Array "stringID of the type" => [
+	 * 					'desc' => "translated string description for the setting"
+	 * 					'methods' => [self::METHOD_*],
+	 * 				]
 	 * @since 8.0.0
+	 * @changed 8.2.0 - Added Array[] support to allow limiting notifications to certain methods
```
Comment (LukasReschke) [inline lib/public/activity/iextension.php]: Me doesn't get that :speak_no_evil:

Comment (nickvergessen) [inline lib/public/activity/iextension.php]: Well the return doc has 2 way, previously only the first one was allowed...

File: lib/public/activity/imanager.php [hunk 3]
```diff
@@ -81,8 +81,13 @@ function registerExtension(\Closure $callable);
...
 	 * Will return additional notification types as specified by other apps
+	 *
 	 * @param string $languageCode
-	 * @return array
+	 * @return array Array "stringID of the type" => "translated string description for the setting"
+	 * 				or Array "stringID of the type" => [
+	 * 					'desc' => "translated string description for the setting"
+	 * 					'methods' => [\OCP\Activity\IExtension::METHOD_*],
+	 * 				]
 	 * @since 8.0.0
```
Comment (MorrisJobke) [inline lib/public/activity/imanager.php]: Missing changed tag ;)

**Comment by schiesbn** [comment]
I know it is not what you changed in this PR but I really have a hard time to understand "Limit notifications about creation and changes to your favorite files (Stream only)". All options is about enabling notifications but this one limit the notifications? In which way? From a user point of view (without looking at the code what happens) I don't understand how this option will change the notifications I receive.

File: lib/public/activity/imanager.php (hunk 3 above)
Comment (nickvergessen) [inline lib/public/activity/imanager.php]: Fixed

**Comment by nickvergessen** [comment]
> I know it is not what you changed in this PR but I really have a hard time to understand "Limit notifications about creation and changes to your favorite files (Stream only)". All options is about enabling notifications but this one limit the notifications? In which way? From a user point of view (without looking at the code what happens) I don't understand how this option will change the notifications I receive.

It will filter the previously named notifications `A new file or folder has been created` and `A file or folder has been changed` and only trigger the notifications, when the file is a favorite files

**Comment by schiesbn** [comment]
hm, quite confusing it it is just a item in the list, don't you think @jancborchardt . Maybe this option should be seperated because it is a  more general setting which filters all the notifications I selected from the rest of the list.

Also why  can I only limit it for the stream? If I want notifications only for favorites then this is most likely also the case for mail notifications.

**Comment by jancborchardt** [comment]
Maybe then it should be the first setting and say

> Limit notifications to **creations** and **changes** in your **favorites** (stream only)

When that is enabled, lots of the stuff below then should be disabled as feedback.

**Comment by nickvergessen** [comment]
Nothing is disabled or can be disabled as every option still manipulates the stream outcome

**Comment by jancborchardt** [comment]
Then I don’t fully get what the function does. It seems half-done and shouldn’t be there hence.

**Comment by schiesbn** [comment]
Why is this for streams only? If I want notifications for favorites only this should affect the stream and the mail notification, right? What are the reasons for making this a stream setting only?

**Comment by nickvergessen** [comment]
Because emails are filtered on input atm and that would make uploading a file too expensive

**Comment by jancborchardt** [comment]
So – do we want to have it in at all in this state? Seems like a half-done thing which we should only introduce when the groundwork is properly finished.

**Comment by nickvergessen** [comment]
Please discuss only stuff from this PR here, the thing you digg out now is completly unrelated and was brought up by me in march.

**Comment by nickvergessen** [comment]
This PR is about making it possible for apps to define if a setting can be used on all notification methods or just some of them and also defines some needed constants. 

If you do not agree that a feature of core is using this method, feel free to open a new issue

**Comment by LukasReschke** [comment]
:+1:

**Pull request merged**